io/fs/ffsb.py

```diff
@@ -0,0 +1,232 @@
+                d[k][2] -= 1
+            if d[k][2] == 0:
+                d[k][2] = 1
+                d[k][3] = d[k][1]
+            # If the ffsb mount point is on the same file system
+            # then use the available disk space after the previous
+            # tests
+            for k1 in d.keys():
+                if d[k1][0] == d[k][0]:
+                    d[k1][1] -= (d[k][2] * d[k][3])
+        fr.close()
+        return d
+
+    def dup_ffsb_profilefl(self):
+        """
+        Validates the path from the FFSB configuration file, the
+        disk space available for the test, warn the user and
+        change the file sizes and/or number of files to be used for
+        generating the workload according to the available disk space
+        on the guest.
+        """
+
+        self.usrfl = '%s/%s' % (os.path.split(self.datadir)[0], 'profile.cfg')
+        self.sysfl = '%s/%s' % (self.datadir, 'profile.cfg')
+
+        params = self.set_ffsb_params(self.usrfl)
+
+        fsno = 0
+        fr = open(self.usrfl, 'r')
+        fw = open(self.sysfl, 'w')
+        for line in fr.read().split('\n'):
+            p = re.compile(r'\s*\t*\[{1}filesystem(\d+)\]{1}')
+            m = p.match(line)
+            if m:
+                fsno = int(line[m.start(1):m.end(1)])
+            p = re.compile(r'(\s*\t*location)\=(.*)')
+            m = p.match(line)
+            if m:
+                while True:
+                    dirnm = ''.join(random.choice(string.letters)
+                                    for i in xrange(9))
+                    if line[m.end(2) - 1] == '/':
+                        newline = '%s%s' % (line[0:m.end(2)], dirnm)
+                        ffsbdir = '%s%s' % (line[m.start(2):m.end(2)], dirnm)
+                    else:
+                        newline = '%s/%s' % (line[0:m.end(2)], dirnm)
+                        ffsbdir = '%s/%s' % (line[m.start(2):m.end(2)], dirnm)
+                    self.tempdirs.append(ffsbdir)
+                    if os.path.exists(ffsbdir):
+                        continue
+                    else:
+                        os.makedirs(ffsbdir)
+                        break
+                fw.write(newline + '\n')
+                continue
+            p = re.compile(r'(\s*\t*num_files)\=(.*)')
+            m = p.match(line)
+            if m:
+                newline = '%s=%s' % (line[0:m.end(1)], str(params[fsno][2]))
+                fw.write(newline + '\n')
+                continue
+            p = re.compile(r'(\s*\t*max_filesize)\=(\d+[kKMmGgTt]?)')
+            m = p.match(line)
+            if m:
+                newline = '%s%s' % (line[0:m.start(2)], str(params[fsno][3]))
+                fw.write(newline + '\n')
+                continue
+            fw.write(line + '\n')
+        fr.close()
+        fw.close()
+
+    def setUp(self):
+        """
+        Build 'ffsb'.
+        """
+
+        self.update_config('profile.cfg.sample')
```

Comment:
This `profile.cfg`  handling is quite confuse. I think you should:
 - Rename `profile.cfg.sample`  to `profile.cfg.default`.
 - Check if a custom `profile.cfg` is provided, using `profile.cfg.default` as default:
~~~
profile_cfg = self.params.get('profile_cfg', default=''profile.cfg.default")
~~~
 - After that, call `self.update_config(profile_cfg)` only once, here in `setUp()`.
 - Drop the `self.update_config(cfg)` call from `test()`.